### PR TITLE
P20 1037 update footer language dropdown to use dsco button component

### PR DESCRIPTION
**Pull request by mgc1194**
### This PR:
- Creates an `I18nDropdown` react component with an associated `storybook` test and an `css` stylesheet.  I18nDropdown consists of a `SimpleDoropdown` embedded in a `form` tag.
- Substitutes `renderI18nDropdown()` byt the `I18nDropdown` component in `SmallFooter`.
- Gets rid of the `i18nDropdown` prop that contained the HTML form.
- Adds new props  `i18nDropdownInBase`, `localeUrl`, and `localeOptions`.
- Removes the i18n_dropdown method from locale_elper used only in SmallFooter.
- Cleans up the usage of `SmallFooter` to include the new props.

The language dropdown is larger to accommodate different languages.

## Links

- spec: [figma](https://www.figma.com/design/liPou77RuNL0Ik7YujsEjN/One-Time-Projects?node-id=1154-1141&node-type=instance&t=0aRVJ2D1BP7HRSKs-0)
- jira ticket: [P20-1037](https://codedotorg.atlassian.net/browse/P20-1037)

## Testing story
| Test Scenario | Screenshot |
| - | - |
| Before - Afrer Flapy | Screenshot 2024-09-30 at 20 30 54  |
|Before - After Aplab | Screenshot 2024-09-30 at 20 32 57 |
| Before - Afrer Minecraft | Screenshot 2024-09-30 at 20 33 12|
| Before - Afrer javalab| Screenshot 2024-09-30 at 20 34 55 |
| Before - After MusicLab | Screenshot 2024-09-30 at 20 35 07 |
| Before - After AI Chat | Screenshot 2024-09-30 at 20 35 44 |

i18n Dropdown in Right-to-Left languages:
Screenshot 2024-09-30 at 20 40 53

Behaviour:

https://github.com/user-attachments/assets/741c99cc-b8e0-4469-818d-3972dff1ff55

## Follow-up work

Use this component in other footers.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -739,25 +739,6 @@ $small-footer-standard-width: 400px;
       background-color: $background-gray;
     }
   }
-
-  .globe-icon {
-    color: $charcoal;
-    font-size: 18px;
-    margin: 0 1px;
-  }
```
Comment (mgc1194) [inline dashboard/app/assets/stylesheets/application.scss:747]: globe-icon is not used anymore. We have a `::before` pseudoelement containing the globe icon.

**Comment by cnbrenci** [comment]
I checked out your branch because I have a related task and was seeing music lab's footer looking too wide on my local- 

image

This is in an allthethings level http://localhost-studio.code.org:9000/s/allthethings/lessons/46/levels/2

It looks like the width: 80px is being overwritten by width: 100% due to the shuffle in components on this level.  Where did you get the music lab screenshot? I'm wondering why it looks different.

File: dashboard/app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -872,25 +853,42 @@ $small-footer-standard-width: 400px;
       .version-caption {
```
Comment (cnbrenci) [inline dashboard/app/assets/stylesheets/application.scss:842]: `max-width: 211px;` on line 853 above (inside the `.small-footer-base` rule will make the footer show up small enough in music lab

image

**Review by cnbrenci — APPROVED** [review]
Looks good pending conflict resolution and merging with staging to re-test & screenshot after https://github.com/code-dot-org/code-dot-org/pull/61410 goes in!

**Pull request merged**